Make spectral anvil can pass through some blocks 使幻灵铁砧可以穿过部分方块

QiuShui1012 opened the pull request:
- resolved #2852

QiuShui1012 committed:
使幻灵铁砧可以穿过部分方块

Gu-ZT reviewed (changes requested):
addDataGenerator 应该是在 common init 调用的

Gu-ZT committed:
refactor(datagen): 移除不必要的注解和优先级设置

- 删除了 `@NotNull` 注解，因为其在当前上下文中不是必需的
- 移除了 `EventPriority.HIGHEST` 设置，使用默认事件订阅优先级
- 删除了未使用的 `AppEngDataCompat` 类
- 为应用工程兼容性添加了区块标签生成器
- 添加了包级别的注解以确保方法和参数非空约束
